### PR TITLE
A couple of logging improvements

### PR DESCRIPTION
**Pull request by bogner**

### CODE REVIEW

**Comment by bogner** [comment]
@swift-ci please test

**Comment by bogner** [comment]
@swift-ci please test

**Comment by bogner** [comment]
@swift-ci please test

**Comment by bogner** [comment]
@swift-ci please test

**Comment by dexonsmith** [comment]
I don’t know what’s going wrong with this one, does it make sense it could be caused by your changes?

https://ci.swift.org/job/llvm-infrastructure-tools-pr-macOS/117/consoleText

**Comment by dexonsmith** [comment]
@swift-ci test

File: src/split2mono.cpp [hunk 1]
```diff
@@ -371,12 +371,25 @@ static int main_check_upstream(const char *cmd, int argc, const char *argv[]) {
 
   // Check if main is up-to-date.
   auto existing_entry = main.upstreams.find(upstream.name);
-  if (existing_entry == main.upstreams.end() ||
-      existing_entry->second.num_upstreams != (long)upstream.upstreams.size() ||
-      existing_entry->second.commits_size != upstream.commits_size_on_open() ||
-      existing_entry->second.svnbase_size != upstream.svnbase_size_on_open()) {
-    fprintf(stderr, "'%s' is not up-to-date with '%s'\n", main.name.c_str(),
-            upstream.name.c_str());
+  auto report_out_of_date = [&](const char *reason) {
```
Comment (dexonsmith) [inline src/split2mono.cpp]: I think it would be better to `return 1` here and simplify the code below.

**Comment by bogner** [comment]
> I don’t know what’s going wrong with this one, does it make sense it could be caused by your changes?
> 
> https://ci.swift.org/job/llvm-infrastructure-tools-pr-macOS/117/consoleText

Doesn't make sense - it's like the test dependencies weren't built

File: src/split2mono.cpp (hunk 1 above)
Comment (bogner) [reply src/split2mono.cpp]: In the lambda?

Comment (dexonsmith) [reply src/split2mono.cpp]: Yup, then you can `return report_out_of_date(...);`.

**Pull request closed without merging**